config/config-api/src/main/java/com/thoughtworks/go/config/PipelineConfig.java

```diff
@@ -367,6 +367,15 @@ public boolean isConfigDefinedRemotely() {
         return this.origin instanceof RepoConfigOrigin;
     }
 
+    public boolean hasSameConfigOrigin(PipelineConfig other) {
+        if (!(isConfigDefinedRemotely()))
+            return false;
+
+        RepoConfigOrigin otherOrigin = (RepoConfigOrigin) other.getOrigin();
+        RepoConfigOrigin repoConfigOrigin = (RepoConfigOrigin) this.origin;
+        return repoConfigOrigin == otherOrigin;
```

Comment:
`==` would check reference equality. You should call `equals`

Reply:
@tomzo You're totally right. I'll push that up real quick